discovery/ec2/ec2.go

```diff
@@ -257,13 +257,20 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
 				if inst.VpcId != nil {
 					labels[ec2LabelVPCID] = model.LabelValue(*inst.VpcId)
 
+					// Deduplicate VPC Subnet IDs maintaining the order of the network interfaces returned by EC2.
+					// The Subnet ID from the instance metadata (if available) will always be first on the list.
```

Comment:
It sounds like the instance metadata subnet should be a separate label, rather than overloading the meaning of this one.

Reply:
The subnet returned in the metadata is the subnet of the primary network interface of the instance. The primary interface cannot be changed or detached from an instance, so the subnet will always be part of the resulting list.

The primary subnet should always be present for VPC instances so the nil check should not be necessary, but I didn't want to risk a panic if there's some edge case where it's missing.

So this PR is not overloading the meaning of the label, just rearranging the list. I can move it to a separate label if you think it's better that way.